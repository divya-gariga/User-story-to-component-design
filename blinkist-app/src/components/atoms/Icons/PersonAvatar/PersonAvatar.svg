<svg width="14" height="19" viewBox="0 0 14 19" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M13.6667 18.3333H12V16.6666C12 16.0036 11.7366 15.3677 11.2678 14.8989C10.799 14.43 10.1631 14.1666 9.50004 14.1666H4.50004C3.837 14.1666 3.20111 14.43 2.73227 14.8989C2.26343 15.3677 2.00004 16.0036 2.00004 16.6666V18.3333H0.333374V16.6666C0.333374 15.5616 0.772361 14.5018 1.55376 13.7204C2.33516 12.939 3.39497 12.5 4.50004 12.5H9.50004C10.6051 12.5 11.6649 12.939 12.4463 13.7204C13.2277 14.5018 13.6667 15.5616 13.6667 16.6666V18.3333ZM7.00004 10.8333C6.34343 10.8333 5.69325 10.704 5.08662 10.4527C4.48 10.2014 3.9288 9.83314 3.46451 9.36885C3.00021 8.90455 2.63192 8.35336 2.38064 7.74673C2.12937 7.1401 2.00004 6.48992 2.00004 5.83331C2.00004 5.1767 2.12937 4.52652 2.38064 3.9199C2.63192 3.31327 3.00021 2.76207 3.46451 2.29778C3.9288 1.83349 4.48 1.46519 5.08662 1.21392C5.69325 0.962642 6.34343 0.833313 7.00004 0.833313C8.32612 0.833313 9.59789 1.3601 10.5356 2.29778C11.4733 3.23546 12 4.50723 12 5.83331C12 7.1594 11.4733 8.43116 10.5356 9.36885C9.59789 10.3065 8.32612 10.8333 7.00004 10.8333ZM7.00004 9.16665C7.8841 9.16665 8.73194 8.81546 9.35706 8.19034C9.98218 7.56521 10.3334 6.71737 10.3334 5.83331C10.3334 4.94926 9.98218 4.10141 9.35706 3.47629C8.73194 2.85117 7.8841 2.49998 7.00004 2.49998C6.11598 2.49998 5.26814 2.85117 4.64302 3.47629C4.0179 4.10141 3.66671 4.94926 3.66671 5.83331C3.66671 6.71737 4.0179 7.56521 4.64302 8.19034C5.26814 8.81546 6.11598 9.16665 7.00004 9.16665Z" fill="#6D787E"/>
</svg>
