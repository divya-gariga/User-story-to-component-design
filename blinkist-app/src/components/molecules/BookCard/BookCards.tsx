import React from "react";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import CardMedia from "@mui/material/CardMedia";
import "./card.css";
import { Time } from "../../atoms/Icons/icons/icons";

interface CardProps {
  img: string;
  title: string;
  author: string;
  readCount: string;
  readTime: string;
  increasedHeight: string;
  personAvatar: string;
}
const BookCards = (props: CardProps) => {

  return (
    <Card
      sx={{
        maxWidth: "282px",
        borderBottom: "1px solid white",
        borderRadius: "8px 8px 0px 0px",
        height: "max-Content",

      }}
    >
      <CardMedia
        component="img"
        image={props.img}

        height={props.increasedHeight} alt="media"
        sx={{ "objectFit": 'contain' }}
      />
      <CardContent>
        <div className="book-card_title">{props.title}</div>
        <div className="book-card_author">{props.author}</div>
        <div className="book-card_stats">
          <div className="book-card_stats-item">
            <span className="icon">
              <Time />
            </span>
            <span className="book-card_readTime" > 13-minutes read</span>
            <span className="book-card_reads">
              <span className="book-card_icon"><img src={props.personAvatar} /></span>
              <span className="book-card_readCount">{props.readCount}</span>
            </span>
          </div>
        </div>
        <div className="ellipsis"><b>...</b></div>
      </CardContent>
    </Card >
  );
};
export default BookCards;
