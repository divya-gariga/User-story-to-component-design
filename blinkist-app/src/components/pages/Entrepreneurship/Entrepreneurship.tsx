import { Box, Grid, Typography } from "@mui/material";
import axios from "axios";
import React, { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import BookCards from "../../molecules/BookCard/BookCards";
import Primary from "../../molecules/Buttons/Primary/Primary";
import SearchBar from "../../molecules/SearchBar/SearchBar";
import Banner from "../../organisms/Banner/Banner";
import PersonAvatar from "../../atoms/Icons/PersonAvatar/PersonAvatar.svg";
import LineProgressBar from "../../atoms/Icons/ProgressBar/LineProgressBar";
import { isExportDeclaration } from "typescript";
function Entrepreneurship() {
  const [blinksInfo, setBlinksInfo] = useState([]);
  const [blinkState, setBlinkState] = useState(false);
  async function onClickOfAddtoLibrary(
    book: {
      coverImage: string;
      title: string;
      author: string;
      state: string;
      isTrending: boolean;
      isJustAdded: boolean;
      isFeatured: boolean;
      readCount: string;
    },
    id: number
  ) {
    await axios
      .put(`http://localhost:8000/BooksInfo/${id}`, {
        ...book,
        state: "reading",
      })
      .then((response) => {
        console.log(response);
      })
      .catch((error) => {
        console.log(error);
      });
    setBlinkState(!blinkState);
  }
  async function getBookInfo() {
    await axios
      .get("http://localhost:8000/BooksInfo")
      .then((response) => {
        const data = response.data;
        setBlinksInfo(data);
      })
  }
  useEffect(() => {
    getBookInfo();
  }, [blinkState]);
  const styles = {
    title: {
      fontFamily: "CeraProBold",
      fontStyle: "normal",
      color: "#03314B",
      fontSize: "24px",
      lineHeight: "30px",
    },
    padding: {
      padding: "100px 18.33% 76px 18.33%",
    },
    link: {
      textDecoration: "none",
    }
  };
  function cardBuilder(book: {
    id: number;
    coverImage: string;
    title: string;
    author: string;
    state: string;
    isTrending: boolean;
    isJustAdded: boolean;
    isFeatured: boolean;
    link: string;
    readCount: string;
  }) {
    let displayProgressBar = "";
    let displayAddToLibrary = "";
    let displayAvatar: any;
    if (book["readCount"] == "") {
      displayAvatar = null;
    }
    else {
      displayAvatar = PersonAvatar;
    }
    let increaseCardHeight = "310px";
    if (book["state"] == "") {
      displayProgressBar = "none";
    } else if (book["state"] != "") {
      displayAddToLibrary = "none";
      increaseCardHeight = "335px";
    }
    function exportId(id: Number) {
      console.log("on clicking" + id);
    }
    return (
      <Grid item pb={"25px"} key={book["id"]} data-testid={`${book["id"]}`}>
        <Box width={"300px"}>
          <Link to={{ pathname: `/Entrepreneurship/${book["id"]}` }} style={styles.link}>
            <div>
              <BookCards
                img={book["coverImage"]}
                title={book["title"]}
                author={book["author"]}
                readCount={book["readCount"]}
                readTime={"13 minutes read"}
                personAvatar={displayAvatar}
                increasedHeight={increaseCardHeight}
              />
            </div>
          </Link>
          <Primary
            label="+ Add to library"
            buttonBorderRadius={{ borderRadius: "0px 0px 8px 8px" }}
            displayElement={{ display: displayAddToLibrary }}
            onClickHandler={() => onClickOfAddtoLibrary(book, book["id"])}
          />
          <LineProgressBar
            displayElement={{ display: displayProgressBar }}
            topHeight={{ top: "32px" }}
            progressValue={35}
          />
        </Box>
      </Grid>
    );
  }
  return (
    <>
      <div style={styles.padding}>
        <Box sx={{ display: "flex", flexDirection: "column" }}>
          <Banner />
          <div
            style={{
              paddingTop: "58px",
              color: "#6D787E",
              fontWeight: "700",
              fontSize: "24px",
              lineHeight: "30px",
              fontStyle: "normal",
            }}
          >
            {/* <SearchBar icon="search" /> */}
          </div>
          <Typography style={styles.title} pt={"80px"} pb={"25px"}>
            Trending Blinks
          </Typography>
          <Grid container columnSpacing={"3%"} data-testid="trendingBlinks">
            {blinksInfo &&
              blinksInfo
                .filter((book) => {
                  return book["isTrending"];
                })
                .map((book) => cardBuilder(book))}
          </Grid>
          <Typography style={styles.title} pb={"25px"} pt={"55px"}>
            Just Added
          </Typography>
          <Grid container columnSpacing={"1%"} data-testid="justAddedBlinks">
            {blinksInfo &&
              blinksInfo
                .filter((book) => {
                  return book["isJustAdded"];
                })
                .map((book) => cardBuilder(book))}
          </Grid>
          <Typography style={styles.title} pb={"25px"} pt={"55px"}>
            Featured Audio Books
          </Typography>
          <Grid container columnSpacing={"1%"} data-testid="featuredBlinks">
            {blinksInfo &&
              blinksInfo
                .filter((book) => {
                  return book["isFeatured"];
                })
                .map((book) => cardBuilder(book))}
          </Grid>
        </Box>
      </div>
    </>
  );
}
export default Entrepreneurship;
