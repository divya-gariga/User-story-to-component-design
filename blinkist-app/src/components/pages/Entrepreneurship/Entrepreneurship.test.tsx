import React from 'react';
import { render, screen, cleanup, fireEvent,waitFor} from '@testing-library/react';
import Entrepreneurship from './Entrepreneurship';
import renderer from 'react-test-renderer';
import PersonAvatar from "../../components/atoms/Icons/PersonAvatar/PersonAvatar.svg";
import BookCards from '../../molecules/BookCard/BookCards';
import '@testing-library/jest-dom'
import {setupServer} from 'msw/node'
import {rest} from 'msw'

const server = setupServer(
  rest.get('http://localhost:8000/BooksInfo', (req, res, ctx) => {
    // respond using a mocked JSON body
    return res(ctx.json({ 
      "id": 2,
      "coverImage": "assets/beyond entrepreneurship 2.png",
      "title": "Beyond Entrepreneurship",
      "author": "Jim Collins & Lazier",
      "state": "finished",
      "isTrending": true,
      "isJustAdded": false,
      "isFeatured": false,
      "link": "/BeyondEntrepreneurship",
      "readCount": "1.9k reads"
    }))
  }),
)
beforeAll(() => server.listen())
afterEach(() => server.resetHandlers())
afterAll(() => server.close())

test('text',()=>{
render(<Entrepreneurship/>)
expect(screen.getByText(/Explore books on entrepreneurship/i)).toBeInTheDocument();
expect(screen.getByText(/Everything you need to know about thriving on a shoestring budget, making your first million, and hiring right from the start./i)).toBeInTheDocument();
expect(screen.getByText(/Trending Blinks/i)).toBeInTheDocument();
expect(screen.getByText(/Just Added/i)).toBeInTheDocument();
expect(screen.getByText(/Featured Audio Books/i)).toBeInTheDocument();
expect(screen.getByTestId("trendingBlinks")).toBeInTheDocument();
expect(screen.getByTestId("justAddedBlinks")).toBeInTheDocument();
expect(screen.getByTestId("featuredBlinks")).toBeInTheDocument();
 })
 test("snapshot for Entrepreneurship page", () => {
    const renderedComponent = renderer.create(<Entrepreneurship />).toJSON()
    expect(renderedComponent).toMatchSnapshot()
})

test("bookcard",()=>{
    render(  <BookCards
        img={"assets/beingboss.png"}
        increasedHeight="310px"
        title={"Being Boss"}
        author={"Kathleen Shannon and Emily"}
        readTime={"13-minutes read"}
        readCount={"16k reads"}
        personAvatar={PersonAvatar}
      />)
      expect(screen.getByText(/Being Boss/i)).toBeInTheDocument();
      expect(screen.getByText(/Kathleen Shannon and Emily/i)).toBeInTheDocument();
      expect(screen.getByText(/13-minutes read/i)).toBeInTheDocument();
    })

    test('loads and display blinks',async()=>{
render(<Entrepreneurship/>)
fireEvent.click(screen.getByRole('button'))
 waitFor(()=>expect(screen.getByText("Being Boss")).toBeInTheDocument())
  })

