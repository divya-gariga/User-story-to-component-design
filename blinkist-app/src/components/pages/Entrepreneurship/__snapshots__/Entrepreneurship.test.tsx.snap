// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for Entrepreneurship page 1`] = `
<div
  style={
    Object {
      "padding": "20px 18.33% 76px 18.33%",
    }
  }
>
  <div
    className="MuiBox-root css-j7qwjs"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-buswvo-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-9 css-b1im03-MuiGrid-root"
      >
        <div
          className="css-1p62k78-MuiStack-root"
        >
          <h2
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            style={
              Object {
                "fontSize": "36px",
                "fontStyle": "normal",
                "fontWeight": 700,
                "lineWeight": "45px",
              }
            }
          >
            Explore books on entrepreneurship
          </h2>
          <p
            className="MuiTypography-root MuiTypography-body1 css-1b23ijh-MuiTypography-root"
            style={
              Object {
                "color": "grey",
                "fontFamily": "CeraProLight",
                "fontSize": "18px",
                "fontStyle": "normal",
                "fontWeight": 400,
                "lineHeight": "23px",
              }
            }
          >
            Everything you need to know about thriving on a shoestring budget, making your first million, and hiring right from the start.
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-3 css-17ss7la-MuiGrid-root"
      >
        <img
          src="GirlStudying.svg"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "color": "#6D787E",
          "fontSize": "24px",
          "fontStyle": "normal",
          "fontWeight": "700",
          "lineHeight": "30px",
          "paddingTop": "58px",
        }
      }
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-hecfuy-MuiFormControl-root-MuiTextField-root"
        data-testid="search"
        style={
          Object {
            "fontFamily": "CeraProMedium",
          }
        }
      >
        <div
          className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1480iag-MuiInputBase-root-MuiInput-root"
          onClick={[Function]}
        >
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
          >
            <span
              className="notranslate"
            >
              â€‹
            </span>
            <img
              alt="search"
              src="Search.svg"
            />
          </div>
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={false}
            id="input-with-icon-textfield"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Search by title or author"
            required={false}
            type="text"
          />
        </div>
      </div>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body1 css-8faadc-MuiTypography-root"
      style={
        Object {
          "color": "#03314B",
          "fontFamily": "CeraProBold",
          "fontSize": "24px",
          "fontStyle": "normal",
          "lineHeight": "30px",
        }
      }
    >
      Trending Blinks
    </p>
    <div
      className="MuiGrid-root MuiGrid-container css-vf96d6-MuiGrid-root"
      data-testid="trendingBlinks"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-1wh0av9-MuiTypography-root"
      style={
        Object {
          "color": "#03314B",
          "fontFamily": "CeraProBold",
          "fontSize": "24px",
          "fontStyle": "normal",
          "lineHeight": "30px",
        }
      }
    >
      Just Added
    </p>
    <div
      className="MuiGrid-root MuiGrid-container css-vf96d6-MuiGrid-root"
      data-testid="justAddedBlinks"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-1wh0av9-MuiTypography-root"
      style={
        Object {
          "color": "#03314B",
          "fontFamily": "CeraProBold",
          "fontSize": "24px",
          "fontStyle": "normal",
          "lineHeight": "30px",
        }
      }
    >
      Featured Audio Books
    </p>
    <div
      className="MuiGrid-root MuiGrid-container css-vf96d6-MuiGrid-root"
      data-testid="featuredBlinks"
    />
  </div>
</div>
`;
